oshi-core/src/main/java/oshi/software/common/AbstractOSThread.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2010 - 2020 The OSHI Project Contributors: https://github.com/oshi/oshi/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.software.common;
+
+import oshi.software.os.OSProcess;
+import oshi.software.os.OSThread;
+
+import java.util.Collections;
+import java.util.List;
+
+public abstract class AbstractOSThread implements OSThread {
+
+    private OSProcess parentProcess;
+
+    public AbstractOSThread(OSProcess process) {
+        this.parentProcess = process;
+    }
+
+    @Override
+    public OSProcess getParentProcess() {
+        return this.parentProcess;
+    }
+
+    @Override
+    public int getProcessID() {
```

Comment:
Hmm... this is a key design question.  Should a thread return its parent process ID or have its own numeric thread ID?  Should we have a separate `getThreadID()` or just return the thread id in the `processID` field?  Interested in your thoughts here, something doesn't sit right with returning the parent ID here.